rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/execution/runner.kt

```diff
@@ -152,18 +152,19 @@ fun getProgram(
     systemInterface: SystemInterface = JavaSystemInterface(),
     programFinders: List<RpgProgramFinder> = defaultProgramFinders
 ): CommandLineProgram {
-    if (systemInterface is JavaSystemInterface) {
-        systemInterface.rpgSystem.addProgramFinders(programFinders)
-        programFinders.forEach {
-            systemInterface.getAllLogHandlers().log(RpgProgramFinderLogEntry(it.toString()))
+    return MainExecutionContext.execute(configuration = systemInterface.getConfiguration() ?: Configuration(), systemInterface = systemInterface) {
+        if (systemInterface is JavaSystemInterface) {
+            systemInterface.rpgSystem.addProgramFinders(programFinders)
+            programFinders.forEach {
+                systemInterface.getAllLogHandlers().log(RpgProgramFinderLogEntry(it.toString()))
+            }
+        } else {
+            // for compatibility with other system interfaces using singleton instance
+            RpgSystem.SINGLETON_RPG_SYSTEM?.addProgramFinders(programFinders)
```

Comment:
Could you give me more information about use of `RpgSystem.SINGLETON_RPG_SYSTEM`?

Reply:
This object, which is shared among all the jariko instances (we have one jariko instance for each executing fun) is helpful only in test environment because it is already ready to use, but in a multi tenant environment, where each tenant has its own library paths, is only dangerous.